test/chainedsysimage.jl

```diff
@@ -0,0 +1,51 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+using Test, Libdl
+
+@testset "empty chained sysimage" begin
+    binariesavailable = false
+    try
+        # TODO - consider using LLVM_full_jll to make this test multiplatform
+        success(`ar --version`)
+        success(`llvm-objcopy --version`)
+        success(`ld --version`)
+        binariesavailable = true
+    catch
+    end
+
+    if !binariesavailable
+        @test_broken false # Not supported for this OS yet
+    else
+        sysimg = Base.unsafe_string(Base.JLOptions().image_file)
+        cd(mktempdir()) do
+            sysoa = replace(sysimg, ".$(Libdl.dlext)" => "-o.a")
+            cp("$sysoa", "sys-o.a", force=true)
+            @test success(`ar x sys-o.a`)
+            rm("data.o")
+            mv("text.o", "text-old.o")
+            @test success(`llvm-objcopy --remove-section .data.jl.sysimg_link text-old.o`)
+
+            source_txt = """
+Base.__init_build();
+module PrecompileStagingArea;
+    using Printf
+end;
+@ccall jl_precompiles_for_sysimage(1::Cuchar)::Cvoid;
+println(0.1, 1, 0x2)
+"""
+
+            @test success(`$(Base.julia_cmd()) --sysimage-native-code=chained --startup-file=no --sysimage=$sysimg --output-o chained.o.a -e $source_txt`)
+            @test success(`ar x chained.o.a`) # Extract new sysimage files
+            @test success(`ld -shared -o chained.$(Libdl.dlext) text.o data.o text-old.o`)
```

Comment:
We now have a `LLD_jll` that we can add as a stdlib to provide LD.

Reply:
Yes, I'll update the test to be system independent, but I'd probably need `LLVM_full_jll` because of `llvm-objcopy`. However, this is only a test and not a dependency for for building Julia itself.